File: cpp/potęga.cpp
/*
 * potęga.cpp
 * a0 = 1
 * a1 = a
 * an = a*...*a (n-czynników) dla n zaw.N+ - {1}
 */


#include <iostream>

using namespace std;

float potega_it(float x, int n){
    float wynik = 1;
    for(int i=0;i<n;i++){
        cout<< i <<endl;
        wynik = wynik*x;
    }
    return wynik;
}

int main(int argc, char **argv)
{
    //pobierz od użytkownika podstawę i wykładnik
    float x = 0;
    int n = 0;
    cout<<"podaj podstawę"<<endl;
    cin>>x;
    cout<<"podaj wykładnik"<<endl;
    cin>>n;
    cout<<"potęga: "<<potega_it(x,n)<<endl;
        
    return 0;
}

File: python/potęga.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potęga.py 
#  

def potega_it(x,n):
    i = 1
    wynik = 1
    while i <= n:
        wynik = wynik *x
        i = i + 1
    return wynik
    
def main (args):
    x = float(input("Podaj podstawę: "))
    n = int(input("Podaj wykładnik: "))
    assert potega(1,1) == 1
    assert potega(2,1) == 2
    assert potega(3,3) == 27
    assert potega(2,2) == 4
    
    
    print ("wynik:" , potega(x,n))
    return 0
    
if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
